app/models/planning_application.rb

```diff
@@ -103,11 +126,52 @@ class PlanningApplication < ApplicationRecord
                            returned], to: :withdrawn, after: proc { |comment|
                                                                update!(cancellation_comment: comment)
                                                              }
+      after do
+        audit("withdrawn", cancellation_comment)
+      end
     end
 
     after_all_transitions :timestamp_status_change # FIXME: https://github.com/aasm/aasm#timestamps
   end
 
+  def audit_created
+    audit("created", nil, Current.user.name)
```

Comment:
should we use just callbacks or just in-line blocks?

Reply:
it is a callback - so this gets called after_create - but I couldn't pass an argument to the hook inline method to get the current user. Have also had to modify this one to make it a little more tolerant to nil users, see updated version

Reply:
(maybe a proc on the after_create hook could achieve the same I just didn't see how)